src/main/java/reposense/RepoSense.java

```diff
@@ -70,7 +70,8 @@ public static void main(String[] args) {
                     cliArguments.getOutputFilePath().toAbsolutePath().toString(),
                     formatter.format(ZonedDateTime.now(cliArguments.getZoneId())),
                     cliArguments.getSinceDate(), cliArguments.getUntilDate(),
-                    cliArguments.isSinceDateProvided(), cliArguments.isUntilDateProvided());
+                    cliArguments.isSinceDateProvided(), cliArguments.isUntilDateProvided(),
+                    TimeUtil::getElapsedTime);
```

Comment:
Can I know why this is necessary, why can't we call this function in `ReportGenerator::generateReposReport`? It seems there is no possibility we will use other time functions in the same place?

Reply:
I chose to pass `TimeUtil::getElapsedTime` as an argument so that the system tests can provide their own timing rather than using `TimeUtil::getElapsedTime`.

In the system tests, I am passing in a fixed elapsed time to `generateReposReport` instead of using `TimeUtil::getElapsedTime`

Reply:
Then, that's fine if there is no other solution.

Reply:
There is another possible solution.

We can change the API of `ReportGenerator` slightly.
Currently, we are calling `generateReposReport` from both `Reposense.java` and `ConfigSystemTest`.

We could break `generateReposReport` into 2 public methods: 

- `generateCommitsAndAuthorshipReport` : this will contain code for generating `commits.json` and `authorship.json`
-  `generateSummary`: this will contain code for generating `summary.json`.

We **will still have** the `generateReposReport` method, but it will simply call `generateCommitsAndAuthorshipReport` first, followed by `generateSummary`.

The elapsed time will only be an argument of `generateSummary` and **not** an argument of `generateReposReport`.

In `RepoSense.java`, we can continue to call `generateReposReport` as per normal.
In `ConfigSystemTest`, we will not call `generateReposReport`.
We will call `generateCommitsAndAuthorshipReport` and `generateSummary`  instead, so that we can pass in a fixed elapsed time.

Reply:
@fzdy1914 should I refactor the solution as such ^ ?

Reply:
I think the current approach is fine. Anyway, the system test is meant to test the system as a whole. There is no reason to test only a part of the code just for this small issue.